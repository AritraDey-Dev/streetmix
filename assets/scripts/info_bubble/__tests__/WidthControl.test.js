/* eslint-env jest */
import React from 'react'
import { fireEvent, cleanup } from 'react-testing-library'
import { renderWithReduxAndIntl } from '../../../../test/helpers/render'

import WidthControl from '../WidthControl'

jest.mock('../../app/routing', () => {})

describe('WidthControl', () => {
  let variantString, type, currentWidth, activeElement, segment
  beforeEach(() => {
    variantString = 'inbound|regular'
    type = 'streetcar'
    currentWidth = 200
    activeElement = 0
    segment = { type, variantString, segmentType: type, id: '1', width: currentWidth, randSeed: 1 }
  })
  afterEach(cleanup)
  it('renders', () => {
    const wrapper = renderWithReduxAndIntl(<WidthControl />, { initialState: { ui: { unitSettings: { resolution: 1, clickIncrement: 1 } }, street: { segments: [segment] } } })
    expect(wrapper).toMatchSnapshot()
  })
  describe('increase width', () => {
    it('increaeses store width', () => {
      const wrapper = renderWithReduxAndIntl(<WidthControl position={activeElement} />, { initialState: { ui: { unitSettings: { resolution: 1, clickIncrement: 1 } }, street: { segments: [segment] } } })
      fireEvent.click(wrapper.getByTitle(/Increase width/i))
      expect(wrapper.store.getState().street.segments[activeElement].width).toEqual(201)
    })
  })
  describe('decrease width', () => {
    it('decreaeses store width', () => {
      const wrapper = renderWithReduxAndIntl(<WidthControl position={activeElement} />, { initialState: { ui: { unitSettings: { resolution: 1, clickIncrement: 1 } }, street: { segments: [segment] } } })
      fireEvent.click(wrapper.getByTitle(/Decrease width/i))
      expect(wrapper.store.getState().street.segments[activeElement].width).toEqual(199)
    })
  })
})
