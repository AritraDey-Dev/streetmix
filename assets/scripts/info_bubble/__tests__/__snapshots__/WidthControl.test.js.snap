// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WidthControl renders 1`] = `
<DocumentFragment>
  <div
    class="non-variant"
  >
    <button
      class="up-down-input-decrement"
      disabled=""
      tabindex="-1"
      title="Decrease width (hold Shift for more precision)"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-minus fa-w-14 "
        data-icon="minus"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M416 208H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h384c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
          fill="currentColor"
        />
      </svg>
    </button>
    <input
      class="up-down-input-element"
      title="Change width of the segment"
      type="text"
      value="0 m"
    />
    <button
      class="up-down-input-increment"
      tabindex="-1"
      title="Increase width (hold Shift for more precision)"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-plus fa-w-14 "
        data-icon="plus"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
          fill="currentColor"
        />
      </svg>
    </button>
  </div>
</DocumentFragment>
`;
